OpenRA.Mods.RA/Effects/GpsDot.cs

```diff
@@ -124,15 +124,18 @@ public void Tick(World world)
 			if (self.Disposed)
 				world.AddFrameEndTask(w => w.Remove(this));
 
-			if (!self.IsInWorld || self.IsDead)
-				return;
-
 			for (var playerIndex = 0; playerIndex < dotStates.Count; playerIndex++)
 			{
 				var state = dotStates[playerIndex];
 				var shouldRender = false;
-				var targetable = (state.Gps.Granted || state.Gps.GrantedAllies) && IsTargetableBy(world.Players[playerIndex], out shouldRender);
-				state.IsTargetable = targetable;
+				var targetable = false;
+
+				if (self.IsInWorld && !self.IsDead)
+				{
+					targetable = (state.Gps.Granted || state.Gps.GrantedAllies) && IsTargetableBy(world.Players[playerIndex], out shouldRender);
+					state.IsTargetable = targetable;
```

Comment:
Since you are here already just assign this to the expression above (instead of using a temporary variable).